commit a34a766ff96d9e88572e35a45066279e40a85d84
Author: Josh Poimboeuf <jpoimboe@kernel.org>
Date:   Mon Jul 24 18:36:58 2017 -0500

    x86/kconfig: Make it easier to switch to the new ORC unwinder
    
    A couple of Kconfig changes which make it much easier to switch to the
    new CONFIG_ORC_UNWINDER:
    
    1) Remove x86 dependencies on CONFIG_FRAME_POINTER for lockdep,
       latencytop, and fault injection.  x86 has a 'guess' unwinder which
       just scans the stack for kernel text addresses.  It's not 100%
       accurate but in many cases it's good enough.  This allows those users
       who don't want the text overhead of the frame pointer or ORC
       unwinders to still use these features.  More importantly, this also
       makes it much more straightforward to disable frame pointers.
    
    2) Make CONFIG_ORC_UNWINDER depend on !CONFIG_FRAME_POINTER.  While it
       would be possible to have both enabled, it doesn't really make sense
       to do so.  So enforce a sane configuration to prevent the user from
       making a dumb mistake.
    
    With these changes, when you disable CONFIG_FRAME_POINTER, "make
    oldconfig" will ask if you want to enable CONFIG_ORC_UNWINDER.
    
    Signed-off-by: Josh Poimboeuf <jpoimboe@redhat.com>
    Cc: Andy Lutomirski <luto@kernel.org>
    Cc: Borislav Petkov <bp@alien8.de>
    Cc: Brian Gerst <brgerst@gmail.com>
    Cc: Denys Vlasenko <dvlasenk@redhat.com>
    Cc: H. Peter Anvin <hpa@zytor.com>
    Cc: Jiri Slaby <jslaby@suse.cz>
    Cc: Linus Torvalds <torvalds@linux-foundation.org>
    Cc: Mike Galbraith <efault@gmx.de>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Cc: live-patching@vger.kernel.org
    Link: http://lkml.kernel.org/r/9985fb91ce5005fe33ea5cc2a20f14bd33c61d03.1500938583.git.jpoimboe@redhat.com
    Signed-off-by: Ingo Molnar <mingo@kernel.org>

diff --git a/arch/x86/Kconfig.debug b/arch/x86/Kconfig.debug
index dc10ec6ead6e..268a318f9439 100644
--- a/arch/x86/Kconfig.debug
+++ b/arch/x86/Kconfig.debug
@@ -357,7 +357,7 @@ config PUNIT_ATOM_DEBUG
 
 config ORC_UNWINDER
 	bool "ORC unwinder"
-	depends on X86_64
+	depends on X86_64 && !FRAME_POINTER
 	select STACK_VALIDATION
 	---help---
 	  This option enables the ORC (Oops Rewind Capability) unwinder for
@@ -365,9 +365,8 @@ config ORC_UNWINDER
 	  a simplified version of the DWARF Call Frame Information standard.
 
 	  This unwinder is more accurate across interrupt entry frames than the
-	  frame pointer unwinder.  It can also enable a 5-10% performance
-	  improvement across the entire kernel if CONFIG_FRAME_POINTER is
-	  disabled.
+	  frame pointer unwinder.  It also enables a 5-10% performance
+	  improvement across the entire kernel compared to frame pointers.
 
 	  Enabling this option will increase the kernel's runtime memory usage
 	  by roughly 2-4MB, depending on your kernel config.